File: Lab9/Tv/TV.py
from TvTuner import TvTuner

class TV(TvTuner):
    def __init__(self):
        self.loudness = 50
        self.switch_tv = False
        self.channel = 0
        self.fout = open("MyTvTuner.txt", "a")

    def get_fout(self):
        return self.fout

    def include(self):
        if not self.switch_tv:
            self.switch_tv = True
            print("TV is on")
            self.get_fout().write("TV is on\n")
        else:
            print("TV was already on")
            self.get_fout().write("TV was already on\n")
        self.get_fout().flush()

    def exclude(self):
        if self.switch_tv:
            self.switch_tv = False
            print("TV is off")
            self.get_fout().write("TV is off\n")
        else:
            print("TV was already off")
            self.get_fout().write("TV was already off\n")
        self.get_fout().flush()

    def change_volume_vol_add(self):
        if self.loudness <= 100:
                print("Volume TV:", self.loudness)
                self.get_fout().write("Volume TV: {}\n".format(self.loudness))
                self.loudness += 1
        else:
            print("Volume TV is maximum")
            self.get_fout().write("Volume TV is maximum\n")
        self.get_fout().flush()

    def change_volume_vol_minus(self):
        if self.loudness >= 0:
            print("Volume TV:", self.loudness)
            self.get_fout().write("Volume TV: {}\n".format(self.loudness))
            self.loudness -= 1
        else:
            print("Volume TV is minimum")
            self.get_fout().write("Volume TV is minimum\n")
        self.get_fout().flush()

File: Lab3/src/lab3/TV.java
package lab3;
import lab3.TvTuner;

import java.io.File;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.PrintWriter;

public class TV implements TvTuner{

    private byte loudness = 50;
    private PrintWriter fout = new PrintWriter(new FileOutputStream(
            new File("MyTvTuner.txt"),
            true));

    private boolean switchTV = false;
    private byte channel = 0;

    public TV() throws FileNotFoundException {
    }

    public PrintWriter getFout(){
        return fout;
    }


    @Override
    public void include() {
        if (!switchTV){
            switchTV = true;
            System.out.println("Телевізор включився");
            getFout().println("Телевізор включився");
        }else {
            System.out.println("Телевізор вже був включений");
            getFout().println("Телевізор вже був включений");
        }
        getFout().flush();
    }

    @Override
    public void exclude() {
        if (switchTV){
            switchTV = false;
            System.out.println("Телевізор виключили");
            getFout().println("Телевізор виключили");

        }else {
            System.out.println("Телевізор вже був виключений");
            getFout().println("Телевізор вже був виключений");

        }
        getFout().flush();

    }


    @Override
    public void change_volume_vol_add() {
        if(loudness<=100){
            System.out.println("Звук: "+(++loudness) );
            getFout().println("Звук: "+(loudness));
        }else{
            System.out.println("Вже стоїть максимальна гучність");
            getFout().println("Вже стоїть максимальна гучність");
        }
        getFout().flush();

    }

    @Override
    public void change_volume_vol_minus() {
        if(loudness>=0){
            System.out.println("Звук: "+(--loudness));
            getFout().println("Звук: "+(loudness));
        }else{
            System.out.println("Вже стоїть мінімальна гучність");
            getFout().println("Вже стоїть мінімальна гучність");
        }
        getFout().flush();
    }

}
